middlewares/retry.go

```diff
@@ -11,24 +11,27 @@ import (
 	"github.com/vulcand/oxy/utils"
 )
 
+// Compile time validation responseRecorder implements http interfaces correctly.
 var (
-	_ http.ResponseWriter = &ResponseRecorder{}
-	_ http.Hijacker       = &ResponseRecorder{}
-	_ http.Flusher        = &ResponseRecorder{}
-	_ http.CloseNotifier  = &ResponseRecorder{}
+	_ http.ResponseWriter = &retryResponseRecorder{}
+	_ http.Hijacker       = &retryResponseRecorder{}
+	_ http.Flusher        = &retryResponseRecorder{}
+	_ http.CloseNotifier  = &retryResponseRecorder{}
 )
 
 // Retry is a middleware that retries requests
 type Retry struct {
 	attempts int
 	next     http.Handler
+	metrics  RetryMetrics
 }
 
 // NewRetry returns a new Retry instance
-func NewRetry(attempts int, next http.Handler) *Retry {
+func NewRetry(attempts int, next http.Handler, metrics RetryMetrics) *Retry {
```

Comment:
Usually I refrain from building abstractions before they are actually required. Therefore personally I would rather leave this part as is, extending it with a more generic implementation once we have an use-case for this. Agree?